src/main/java/com/googleinterns/zoomtube/servlets/IconFeedbackServlet.java

```diff
@@ -62,6 +67,38 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     datastore.put(createEntityFromRequest(request));
   }
 
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Optional<String> getRequestError = validateGetRequest(request);
+    if (getRequestError.isPresent()) {
+      response.sendError(HttpServletResponse.SC_BAD_REQUEST, getRequestError.get());
+      return;
+    }
+
+    List<IconFeedback> lectures = getIconFeedbacks(request);
+    Gson gson = new Gson();
+    response.setContentType("application/json");
+    response.getWriter().println(gson.toJson(lectures));
+  }
+
+  /** Returns IconFeedbacks stored in the database. */
```

Comment:
Can you mention it is for a specific lecture?

Reply:
added